lua/core/menu.lua

```diff
@@ -1145,7 +1146,104 @@ m.deinit[pRESET] = function() end
 
 
 -----------------------------------------
+-- UPDATE
+m.update = {}
+m.update.url = ''
+m.update.version = ''
+
+local function check_newest()
+  print("checking for update")
+  m.update.url = util.os_capture( [[curl -s \
+      https://api.github.com/repos/monome/norns/releases/latest \
+      | grep "browser_download_url.*" \
+      | cut -d : -f 2,3 \
+      | tr -d \"]])
+  print(m.update.url)
+  m.update.version = m.update.url:match("(%d%d%d%d%d%d)")
+  print("available version "..m.update.version)
+end
+
+local function get_update()
+  m.update.message = "preparing..."
+  menu.redraw()
+  pcall(cleanup) -- shut down script
+  norns.script.clear()
+  os.execute("sudo systemctl stop norns-jack.service") -- disable audio
+  os.execute("sudo rm -rf /home/we/update/*") -- clear old updates
+  m.update.message = "downloading..."
+  menu.redraw()
+  os.execute("wget -q -P /home/we/update/ " .. m.update.url) --download
+  m.update.message = "unpacking update..."
+  menu.redraw()
+  os.execute("tar xzvf /home/we/update/*.tgz -C /home/we/update/")
+  m.update.message = "running update..."
+  menu.redraw()
+  os.execute("/home/we/update/"..m.update.version.."/update.sh")
+  m.update.message = "complete."
+  menu.redraw()
+end
+
+m.key[pUPDATE] = function(n,z)
+  if m.update.stage==1 and z==1 then
+    menu.set_page(pSYSTEM)
+    menu.redraw()
+  elseif m.update.stage==2 then
+    if n==2 and z==1 then
+      menu.set_page(pSYSTEM)
+      menu.redraw()
+    elseif n==3 and z==1 then
+      m.update.stage=3
+      get_update()
+      m.update.stage=4
+    end
+  elseif m.update.stage==4 and z==1 then
+    print("shutting down.")
+    m.update.message = "shutting down."
+    menu.redraw()
+    os.execute("sleep 0.5; sudo shutdown now")
+  end
+end
 
+
+m.enc[pUPDATE] = function(n,delta) end
+
+m.redraw[pUPDATE] = function()
+  screen.clear()
+  screen.level(15)
+  screen.move(64,40)
+  if m.update.stage == 1 then
+    screen.text_center(m.update.message)
+  elseif m.update.stage == 2 then
+    screen.text_center("update found: "..m.update.version)
+    screen.move(64,50)
+    screen.text_center("install?")
+  elseif m.update.stage == 3 then
+    screen.text_center(m.update.message)
+  end
+  screen.update()
+end
+
+m.init[pUPDATE] = function()
+  m.update.stage = 1
+
+  local ping = util.os_capture("ping -c 1 github.com | grep failure")
+  if ping == '' then check_newest() end
+
+  if not ping == ''  then
+    m.update.message = "need internet."
+  elseif tonumber(norns.version.update) >= tonumber(m.update.version) then
+    m.update.message = "up to date."
+  elseif norns.disk < 400 then
+    m.update.message = "disk full."
```

Comment:
i'd be tempted to add more detail (like how much space is required), but maybe there's not enough screen real estate?